actors/builtin/market/market_test.go

```diff
@@ -971,6 +958,52 @@ func TestOnMinerSectorsTerminate(t *testing.T) {
 
 	})
 
+	t.Run("terminating a deal the second time does not change it's slash epoch", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		rt.SetEpoch(currentEpoch)
+
+		dealId1 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch)
+		actor.activateDeals(rt, sectorExpiry, provider, currentEpoch, dealId1)
+
+		// terminating the deal so slash epoch is the current epoch
+		actor.terminateDeals(rt, provider, dealId1)
+
+		// set a new epoch and terminate again -> however slash epoch will still be the old epoch.
+		newEpoch := currentEpoch + 1
+		rt.SetEpoch(newEpoch)
+		actor.terminateDeals(rt, provider, dealId1)
+		st := actor.getDealState(rt, dealId1)
+		require.EqualValues(t, currentEpoch, st.SlashEpoch)
+	})
+
+	t.Run("terminating new deals and an already terminated deal only terminates the new deals", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		rt.SetEpoch(currentEpoch)
+
+		// provider1 publishes deal1 and 2 and deal3 -> deal3 has the lowest endepoch
+		dealId1 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch)
+		dealId2 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch+1)
+		dealId3 := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch-1)
+		actor.activateDeals(rt, sectorExpiry, provider, currentEpoch, dealId1, dealId2, dealId3)
+
+		// terminating the deal so slash epoch is the current epoch
+		actor.terminateDeals(rt, provider, dealId1)
+
+		// set a new epoch and terminate again -> however slash epoch will still be the old epoch.
```

Comment:
nit: slash epoch for dealID1 will still be the old epoch

Reply:
Will do this in a follow-up.